---
title: 'Your Title Here'
author: "Kim Medina"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: html_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable.csv"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = FALSE)
```

# Background and Overview

*Plasmodium falciparum* is a unicellular parasite known to cause malaria in humans. While it is not the sole source of malaria, with *Plasmodium vivax* also a great contributor to the disease, it is the deadliest species of *Plasmodium*. This parasite is transmitted through the bite of a female Anopheles mosquito and when a human is infected they become sick with malaria [@whoMalaria]. Malaria is incredibly deadly as it can cause organ failure, anemia, pulmonary edema, among other illnesses. According to the World Health Organization, in 2021 there were an estimated 348,428 cases of malaria in Equatorial Guinea with 785 deaths and it remains a major public health problem in the area [@whoEGdata]. Due to its prevalence, it is a significant cause of death for children under the age of 5 with a mortality rate of 77 per 1000 live births. 

While malaria had been a long-standing issue in the region, early 2020 had marked the start of the COVID-19 pandemic, with the SARS-CoV-2 virus causing the disease and adding an extra burden on the healthcare system of Equatorial Guinea. According to worldometer, which tracks Coronavirus cases, by the end of 2021 there were 13,637 positive cases of SARS-CoV-2 in the area. This is significantly less than *Plasmodium falciparum* infections however this number may be underreported due to lack of accessibility to healthcare and thus lack of testing of the population. 

Having both diseases present in an area where healthcare can be inaccessible and hard to get to, co-infections are a likely occurrence. However, according to the study by López-Farfán, et al., only 6 out of 1,556 cases were found to be co-infected with both *Plasmodium falciparum* and SARS-CoV-2 during the course of their study. They also reported that these co-infections were mild. Of the 1,556 cases, 43 genomes of SARS-CoV-2 isolates were sequences, most of which belonged to the Delta variant. The primary goal of this report was with the 43 genome sequences to explore how prevalent SARS-CoV-2 is in Equatorial Guinea in different identity groups such as sex or age and if any identity group was more susceptible to infection in an area where healthcare may not be the most accessible and other illnesses such as malaria are long-standing issues. 

After downloading the metadata from the National Center for Biotechnology Information (NCBI) BioProject database, I used various bash-script commands to explore the dataset and different variables it contained to help achieve the primary goal of this report. R script and code was then used to create the figures in this report to help better visualize the data. 

After doing these analyses, I found that women in their late 20’s were more likely to be infected with SARS-CoV-2 in Equatorial Guinea. However, due to inaccessibility to healthcare and the small sample size associated with the dataset, this conclusion may not be the most accurate as people from older age groups may not have been able to volunteer for this study and thus their data was not added to the dataset.


# Methods

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

You may also want to use any of a range of different COVID data packages and data sources:

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`

## Subsections are ok too

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")
library("tidyr")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <-
  add_genes_metadata_to_vcfstack(sra_runtable_path = params$sra_runtable_path,
                                 stacked_vcf = stacked_vcfs,
                                 cleaned_genes_table = gene_table)
```

# Figures

```{r sample-plot}
# create a plot of unique SNP locations within each gene across all samples
vcf_with_metadata %>%
  filter(!is.na(gene)) %>% # get rid of SNPs not in gene regions
  group_by(gene, pos) %>%
  tally() %>% # this gives a column n for the number of gene by position
  group_by(gene) %>%
  tally() %>% # this collapses that down to the number of unique SNP locations
  ggplot(aes(x = gene,
             y = n)) +
  geom_col() +
  labs(title = "Count of distinct SNPs in Named SARS-CoV-2 Genes",
       x = "Gene Name") +
  theme_few() # get rid of the grey background
```

**Figure 1**: N and S genes have more unique SNPs in the set of samples analyzed.

# Tables

```{r example-table}
# An example table to show the length of each gene using its start and end
gene_table %>%
  mutate(length = end - start) %>%
  select(gene_name, start, end, length) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Start",
                             "End",
                             "Length"))
```

```{r, average-age-of-diseased}
diseased_only <- vcf_with_metadata %>%
  filter(host_health_state == "diseased")
ggplot(diseased_only, aes(x = host_health_state, y = Host_age)) +
  geom_boxplot() +
  theme_minimal() +
  labs(
    title = "Comparison of Host Age by Disease Status",
    x = "Disease Status",
    y = "Host Age"
  )
```

```{r, sex-vs-host-health-state}
sra_run_table <- read.csv("data/00_sra_runtable/SraRunTable.csv")
diseased_only <- sra_run_table %>%
  filter(host_health_state == "diseased")
ggplot(diseased_only, aes(x = host_sex)) +
  geom_bar(fill = "skyblue") +
  theme_minimal() +
  labs(
    title = "Comparison of Host Sex by Count",
    x = "Host Sex",
    y = "Count"
  )

```

```{r, plot-alt-across-genome}
ggplot(vcf_with_metadata, aes(x = pos,y = alt)) +
  geom_point(size = 1.5) +
  labs(
    title = "Alternate SNPs Across the Genome",
    x = "Genomic Position",
    y = "Chromosome")
```

```{r, plot-qual-across-genome}
ggplot(vcf_with_metadata, aes(x = pos, y = as.numeric(qual))) +
  geom_point(size = 1.5) +
  labs(
    title = "SNP Quality of Sample Across SARS-Cov-2 Genome",
    x = "Genomic Position",
    y = "Quality score")
```

```{r, plot-malaria-vs-non-infected}
infection_data <- read.csv("data/raw_data/malaria_infection_in_equatorial_guinea.csv")
malaria_infection_only <- infection_data %>%
  mutate(Non_Infected = subtotal - malaria_infection)

malaria_infection_only_long <- malaria_infection_only %>%
  gather(key = "Status", value = "Count", malaria_infection, Non_Infected)

ggplot(malaria_infection_only_long, aes(x = Status, y = Count, fill = Status)) +
  geom_bar(stat = "identity", position = "stack") +  
  labs(
    title = "Plasmodium falciparum Infections vs Grand Total (Non-Plasmodium 
falciparum Infected)",
    x = "Category",
    y = "Count",
    fill = "Status"
  )
```

```{r, plot-covid-vs-non-infected}
infection_data <- read.csv("data/raw_data/malaria_infection_in_equatorial_guinea.csv")
covid_infection_only <- infection_data %>%
  mutate(Non_Infected = subtotal - sars_cov_2_infection)

covid_infection_only_long <- covid_infection_only %>%
  gather(key = "Status", value = "Count", sars_cov_2_infection, Non_Infected)

ggplot(covid_infection_only_long, aes(x = Status, y = Count, fill = Status)) +
  geom_bar(stat = "identity", position = "stack") +  
  labs(
    title = "SARS-CoV-2 Infections vs Grand Total (Non-SARS-CoV-2 Infected)",
    x = "Category",
    y = "Count",
    fill = "Status"
  )
```


**Table 1**: Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

# Sources Cited
